internal/cache/util/util.go

```diff
@@ -99,3 +101,25 @@ func IsCacheHandleInvalid(readErr error) bool {
 		strings.Contains(readErr.Error(), ErrInSeekingFileHandleMsg) ||
 		strings.Contains(readErr.Error(), ErrInReadingFileHandleMsg)
 }
+
+func CheckCacheDirectoryPermissions(dirPath string) (bool, error) {
+	_, statErr := os.Stat(dirPath)
+
+	if statErr != nil {
+		err := os.MkdirAll(dirPath, FileDirPerm)
```

Comment:
We are still creating cache directory in [cache_handler](https://github.com/GoogleCloudPlatform/gcsfuse/blob/aeb8286b98c70cb02c323a0198ed454a85fc0ab4/internal/cache/file/cache_handler.go#L69). Just wanted to confirm that this is as per design? :)

Reply:
yes we are still creating it, it might be redundant now, we can remove it in future as tech debt, @sethiay please correct me in case I am mistaken

Reply:
it's redundant but not tech debt because the CreateFile call inside the cache_handler is still required and so is the MkdirAll call inside the CreateFile (because the cahe path for an object is like <user passed path> + gcsfuse-file-cache + <bucket_name> + <object_name> where object name can also be of type a/b/c. So MkdirAll is required for path after gcsfuse-file-cache as well.

Reply:
got it ..